Fix inconsistent button styling with TT3

gigitux opened the pull request:
The WooCommerce buttons are styled differently from the current theme buttons on TT3. The issue is reproducible only if the link has never been visited. An easy way to test it is to visit that page from a private browsing window.


<!-- Reference any related issues or PRs here -->

Fixes #7388

<!-- Don't forget to update the title with something descriptive. -->
<!-- If you can, add the appropriate labels -->

### Screenshots

<!-- If your change has a visual component, add a screenshot here. A "before" screenshot would also be helpful. -->

| Before | After |
| ------ | ----- |
|![image](https://user-images.githubusercontent.com/4463174/198559087-b24ceeb2-61ba-488f-84a0-f87d6b2bc965.png)|![image](https://user-images.githubusercontent.com/4463174/198559394-b3e377d5-3e6e-4f3e-9ea4-282fd02fe923.png)|

### Testing


#### User Facing Testing

1. Enable TT3.
2. Create a new post/page and add the `All Product` block and the `Product Categories` block.
3. Save.
4. Visit the post/page via incognito mode and be sure that all the buttons have the same style. 



* [ ] Do not include in the Testing Notes <!-- Check this box if this PR can't be tested by users (ie: it doesn't include user-facing changes or it can't be tested without manually modifying the code). -->

### WooCommerce Visibility

<!-- Check this [this doc](../docs/blocks/feature-flags-and-experimental-interfaces.md) to see if the change is visible in WC core or not (part of the feature plugin or experimental)-->

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### Performance Impact

<!-- Please document any known performance impact (positive or negative) here. If negative, provide some rationale for why this is an okay tradeoff or how this will be addressed. -->

### Changelog

> Fix inconsistent button styling with TT3

gigitux committed:
fix inconsistent button styling with TT3

Aljullu commented on src/BlockTypes/AbstractProductGrid.php:
```diff
@@ -647,7 +647,7 @@ protected function get_add_to_cart( $product ) {
 			'data-product_sku' => $product->get_sku(),
 			'rel'              => 'nofollow',
-			'class'            => 'wp-block-button__link add_to_cart_button',
+			'class'            => 'wp-block-button__link wp-element-button add_to_cart_button',
```
Ideally, we should use `wp_theme_get_element_class_name( 'button' )` to get the correct class name, instead of directly using `wp-element-button`. Similar to what WC core does here: https://github.com/woocommerce/woocommerce/pull/35306/files#diff-3582a823c079f0c8354f3171ce608961ae6b224dd0182f7e4bb14e82278b05abR527-R543.

I don't know if there is an equivalent one for the JS side, but might be good investigating as well.

gigitux replied:
Cool! Thanks for sharing this! I used `wc_wp_theme_get_element_class_name` to be more consistent with WC core [a00e4e6](https://github.com/woocommerce/woocommerce-blocks/pull/7516/commits/a00e4e60311b4fa407787a9451c55aab820f4537)

I searched, but I didn't find any JS equivalent. `wp_theme_get_element_class_name` reads data from theme.json that isn't accessible on the JS side.
https://github.com/WordPress/WordPress/blob/44881d209b2f37cccdbd1f9a8713ba782bbdb65f/wp-includes/theme.php/#L4231-L4233

Aljullu reviewed (commented):
Confirming it fixes the issue, thanks for picking this up, @gigitux! I just left one comment about using a function to retrieve the class name.

Also, I'm not sure about adding the `skip-changelog` label, I think it's worth mentioning it in the changelog.

gigitux committed:
use wc_wp_theme_get_element_class_name

gigitux commented:
> Confirming it fixes the issue, thanks for picking this up, @gigitux! I just left one comment about using a function to retrieve the class name.
> 
> Also, I'm not sure about adding the `skip-changelog` label, I think it's worth mentioning it in the changelog.

Thanks for your review! I addressed your feedback, and I added a changelog entry too.

Aljullu commented on src/BlockTypes/AbstractProductGrid.php:
```diff
@@ -647,7 +647,7 @@ protected function get_add_to_cart( $product ) {
 			'data-product_sku' => $product->get_sku(),
 			'rel'              => 'nofollow',
-			'class'            => 'wp-block-button__link add_to_cart_button',
+			'class'            => 'wp-block-button__link ' . wc_wp_theme_get_element_class_name( 'button' ) . ' add_to_cart_button',
```
If we use this function directly, we should update [the `WC requires at least` version to 7.0](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/woocommerce-gutenberg-products-block.php#L12), which is when `wc_wp_theme_get_element_class_name()` will be introduced. Another option would be to add a check around it to make sure it exists before calling it.

gigitux replied:
> Pre-approving, but left a comment about adding a check around the PHP function or raising the required WC version (which we should do anyway in the next release).

Thanks for the review and thanks for the great catch! I added a check around the PHP function. I'm going to merge it!

Aljullu reviewed (approved):
Pre-approving, but left a comment about adding a check around the PHP function or raising the required WC version (which we should do anyway in the next release).

gigitux committed:
add check to be sure that wc_wp_theme_get_element_class_name function exists

gigitux committed:
Merge branch 'trunk' into fix/add_to_cart_button_TT3
